src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComActivator.cs

```diff
@@ -177,6 +197,72 @@ private static Type FindClassType(Guid clsid, string assemblyName, string typeNa
             throw new COMException(string.Empty, CLASS_E_CLASSNOTAVAILABLE);
         }
 
+        private static AssemblyLoadContext GetALC(string assemblyPath)
+        {
+            AssemblyLoadContext alc;
+            return LazyInitializer.EnsureInitialized(ref s_AssemblyLoadContexts).TryGetValue(assemblyPath, out alc) ?
+                alc:
+                AddNewALC(assemblyPath);
+
+            AssemblyLoadContext AddNewALC(string path)
+            {
+                // The static collection of ALCs is readonly. If a new ALC entry needs to be added,
+                // then we must make a copy, add the new ALC, and atomically update the static member.
+                while (true)
+                {
+                    Dictionary<string, AssemblyLoadContext> prevContexts = s_AssemblyLoadContexts;
```

Comment:
Does the dictionary need to have case-insensitive hash function; or do we expect the path here to be perfectly normalized for case?

Reply:
We will create the paths the same way each time so they should be normalized to be identical, but the cost to use a case-insensitive hash is probably not enough to risk it. Do you have a feeling either way?